sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1466,7 +1475,20 @@ case class RebalancePartitions(
   def partitioning: Partitioning = if (partitionExpressions.isEmpty) {
     RoundRobinPartitioning(conf.numShufflePartitions)
   } else {
-    HashPartitioning(partitionExpressions, conf.numShufflePartitions)
+    val (sortOrder, nonSortOrder) = partitionExpressions.partition(_.isInstanceOf[SortOrder])
+    require(sortOrder.isEmpty || nonSortOrder.isEmpty,
```

Comment:
We can probably have a common trait for this and `RepartitionByExpression`, to share code.